INTERFACE.md

```diff
@@ -0,0 +1,129 @@
+# Interfaces Overview for base-repo
+
+This document aims to answer questions on how to configure external dependencies and which public interfaces are offered by base-repo in a comprehensive way. 
+It is meant to be used for getting an overview and guidance in addition to the official documentation, which is available at the official [base-repo Web page](https://kit-data-manager.github.io/webpage/base-repo/).
+
+> **Note:**
+> This document applies to the base-repo version it is shipped with. If you have a specific version running, please refer to `INTERFACE.md` of this particular release.
+
+## TOC
+
+- [Interfaces Overview for base-repo](#interfaces-overview-for-base-repo)
+   * [External Dependencies](#external-dependencies)
+      + [Relational Database (mandatory)](#relational-database-mandatory)
+         - [Configuration](#configuration)
+      + [Local Filesystem (mandatory)](#local-filesystem-mandatory)
+         - [Configuration](#configuration-1)
+      + [Messaging (optional)](#messaging-optional)
+         - [Configuration](#configuration-2)
+      + [Enhanced Search (optional)](#enhanced-search-optional)
+         - [Configuration](#configuration-3)
+      + [Access Control (optional)](#access-control-optional)
+         - [Configuration](#configuration-4)
+   * [Public Interfaces](#public-interfaces)
+      + [HTTP / REST](#http--rest)
+         - [Documentation](#documentation)
+         - [Application Examples    ](#application-examples)
+      + [Elasticsearch Proxy](#elasticsearch-proxy)
+         - [Documentation](#documentation-1)
+         - [Application Examples](#application-examples-1)
+      + [OAI-PMH](#oai-pmh)
+         - [Source Code](#source-code)
+         - [Documentation](#documentation-2)
+      + [Digital Object Interface Protocol (DOIP)](#digital-object-interface-protocol-doip)
+         - [Source Code](#source-code-1)
+         - [Documentation](#documentation-3)
+
+
+## External Dependencies
+
+External dependencies are third-party services that Qare required for base-repo to work properly or that can be added optionally to provide additional functionality. Typically, external dependencies require
+additional software to be installed and configured, before they can be included in the base-repo configuration, which is typically done via the main configuration file `application.properties`.
+
+### Relational Database (mandatory)
+A relational database is required by base-repo to store administrative metadata for resources and content information. If not configured properly, base-repo will fail to start.
+
+#### Configuration
+ - H2 In-Memory (driver included, used for testing, not recommended for production) [Example](https://github.com/kit-data-manager/base-repo/blob/4e90c6aeaced4715d419482f3cb127cddc85bd37/src/test/resources/test-config/application-test.properties#L31-L34)
```

Comment:
For better readability it may be better to use a tag instead of the id of the commit.
e.g.: https://github.com/kit-data-manager/base-repo/blob/v1.4.0/src/test/resources/test-config/application-test.properties#L31-34
Cons:
May cause problems when referencing future tags.